File: moderate/98_point_in_circle.c
#include <stdio.h>
#include <math.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    float centre_x, centre_y, radius, point_x, point_y;

    while (fgets(line, 1024, file)) {
        sscanf(line, "Center: (%f, %f); Radius: %f; Point: (%f, %f)",
                &centre_x, &centre_y, &radius, &point_x, &point_y);

        /* calculate distance from the point to the centre */
        float distance = sqrt(pow(point_x - centre_x, 2) +
                              pow(point_y - centre_y, 2));

        /* point is inside the circle if it is within the radius */
        printf("%s\n", (distance <= radius) ? "true" : "false");
    }

    return 0;
}

File: moderate/98_point_in_circle.py
import sys
import string
import math

for line in open(sys.argv[1]):
    # strip out unwanted characters and convert numbers to floats
    data = [float(x) for x in
                line.translate(None, string.ascii_letters + ':(,);\n').split()]
    centre = data[0:2]
    radius = data[2]
    point = data[3:5]

    # calculate distance from the point to the centre
    distance = math.sqrt(((point[0]-centre[0]) ** 2) +
                         ((point[1]-centre[1]) ** 2))

    # point is inside the circle if it is within the radius
    print(str(distance <= radius).lower())
